api/autopilot.go

```diff
@@ -146,6 +147,10 @@ func (hgb HostGougingBreakdown) DownloadGouging() bool {
 	return hgb.V3.DownloadErr != ""
 }
 
+func (hgb HostGougingBreakdown) PruneGouging() bool {
+	return hgb.V2.PruneErr != ""
```

Comment:
Shouldn't both this and `DownloadGouging` also check the base `GougingErr` ? 

I also think we could simplify things a bit by getting rid of `V2` and `V3` and instead add that as context to the errors. Otherwise we are going to add `V4` soon and end up with a pretty confusing API.

Reply:
If I remember it correctly, the reasoning was that `GougingErr` includes things like `MaxStoragePrice` and `MaxContractPrice` checks, which don't have anything to do really from the context of a sector download. Therefor we check for gouging "in general" when we form and renew contracts, but the checks performed by the worker check the individual sub checks that zoom in more on what action is being performed at the time. This to ensure a download (might be a migration) wouldn't fail on contract price gouging errors.

> I also think we could simplify things a bit by getting rid of V2 and V3 and instead add that as context to the errors. Otherwise we are going to add V4 soon and end up with a pretty confusing API.

Are you proposing I do that in this PR? I don't really mind but adding the `PruneErr` to me doesn't trigger a gouging refactor 😛 even though I hate our gouging checks

Reply:
I would honestly just call `.Check().Gouging()` instead of narrowing it down to a sub -check. Because a gouging host won't have its contract renewed and if we don't renew a contract, there is no point in pruning.

It can also be a f/u. But just to be clear I'm not talking about a full upload-code-style refactor. Just merging these 2 fields^^